### ISSUE TITLE
2.4.11: How to calculate exactly AND 8 pixels along the shortest side PROBLEM 

### ISSUE THREAD

**Issue by jake-abma** [w3c/wcag#1531]
From: https://broken--deque-workshop.netlify.app/ => the dialogs (2x) for EDIT and RECIPES BLUE BUTTON

**See the following focus styles for the buttons at the footer of the dialogs... ("SAVE" white bracket like thingy on blue background):**

<img width="504" alt="Screen Shot 2020-11-17 at 2 05 16 PM" src="https://user-images.githubusercontent.com/6384002/99393665-f1876d80-28dd-11eb-833d-9d4fc53a548d.png">

**And then a wider one (I COOKED IT)**

<img width="528" alt="Screen Shot 2020-11-17 at 2 03 20 PM" src="https://user-images.githubusercontent.com/6384002/99393687-f8ae7b80-28dd-11eb-8bba-7dee563f4d0d.png">

- To calculate if they pass I started with the 'save' focus and came to the conclusion to draw a 3x15px box and approximately has two of them to have a full 'bracket thingy' = 150px area.
- I have one left and one right so 300px total. 
- The button itself is 110x36 * 2 = 292px
- So, I say it's a PASS

For the second image though, the width is wider and I do not have to calculate it, it's a FAIL

"a thickness of at least 8 CSS pixels along the shortest side" does not apply here (while we should allow, maybe...) 

This is (still) hard to explain!

**Comment by jake-abma** [comment]
@alastc can you take a look here?

short comment / result, when using this demo page for training purposes, we pass the first one (SAVE) and fail the second one (I COOKED IT)

this raises eyebrows with the students

**Comment by patrickhlauke** [comment]
where mathematical models hit real-world content, there's going to be more oddities like this where instinctively we can say something is ok, but it fails the model on a formality... particularly in situations where pass/fail depends solely on the relationship to the overall dimensions (and for extra fun, situations where the size may not be known/depends on other factors ... think for instance a button whose text depends solely on something like the name/title of a product or similar ... you may have a focus indicator that passes for short product titles but fails for long ones, and no way to make sure when/if it will fail)

**Comment by bruce-usab** [comment]
> this raises eyebrows with the students

As well it should IMHO.  I find this pretty compelling evidence that the SC is cooked.

**Comment by alastc** [comment]
A few points come to mind:

1. The second part of the bullet (8px wide) was not intended to another area calculation, but a straight test. 
2. The reverse of that situation is that a fixed-size is allowed, and then you have large focus areas where you cannot track where the focus is because it is so small proportionally.
3. We created an exception for a particular style of indicator, this (one on each side) is another style but same basic problem.
4. The example is from a page that is _intentionally_ bad from an accessibility point of view. I'm not sure if that focus style was intended to be bad, but it wouldn't surprise me. It's an odd style, I'm not sure I'd catch that whilst tabbing through.

If I were providing guidance on making a visible indicator (regardless of the calculations proposed in 2.2), putting something on the long edge (or all the way around) is a more reliable way of making it easy to see. It also avoids the visual confusion where the focus indicator could apply to side by side horizontal links.

I can see a few options, broadly:
- Add more exceptions;
- Remove the exception and keep it simpler (but more prescriptive);
- Give up on defining visibility in 2.2.

**Comment by mraccess77** [comment]
I'd also like to know what happens if there is a border-radius and the 8px method is used - would that pass?  For example border-left is 8px but border-radius is 25px;

**Comment by alastc** [comment]
Hi folks,

This one has been bubbling in my mind for a while, and I'd like to propose changing the size metric to have two area methods, instead of one area method and an exception (of sorts).

I'm proposing:
<hr>

**Minimum area:** The area of the focus indicator is either:
- at least as large as the area of a 1 CSS pixel thick perimeter of the unfocused control;
- at least as large as a 4 CSS pixels border along the shortest side, and no thinner than 2 CSS pixels

<hr>

The rational behind it being 4px is from testing various shapes and trying to come up with an equivalent to the 1px perimeter. If you take a square, the perimeter is (within 5px) the area of one side * 4px.

If you take a narrow element (e.g. 20 x 150px), you have a smaller visual area, but if it is thicker that helps.

It could have opened up a gap because if you only consider the area and not the thickness, the second method means quite a small area for a narrow control. E.g. a 20px by 150px control would have a min area of 80px, which as a 1px line is not good. However, if that area has to be at least 2px thick, the provides better visibility.

Have a look at the [table of sizes and examples](https://alastairc.uk/tests/wcag22-examples/focus-more-visible-5.html). 

I'll pop this on the survey for next week.

**Comment by mraccess77** [comment]
@alastc Since this is area "•	at least as large as a 4 CSS pixels border along the shortest side, and no thinner than 2 CSS pixels" and doesn't specify where it needs to be - I could calculate the shortest side - say 20 pixels and that means my area of focus indicators needs to be 80 and at least 2 pixels wide.  So I could then have a horizontal line at the bottom that was 40 pixels wide and 2 pixels high based on this.    If I understand this correctly this might be a preferred solution.  Then what happens with the 2px wide clause related to contrast - would I not need to meet contrast because it's 2px wide or because it's not the 2px order I still need to meet contrast?  The challenge is then if we drop the need for contrast yet we no longer have a 2px border that is all around/sticks out it is less obvious of a change for low vision users.  Of course then SC 1.4.11 might still come into force and require adjacent contrast anyway?  This gets very confusing quickly in how it might still relate to SC 1.4.11 - but I do like this idea.

**Comment by alastc** [comment]
> So I could then have a horizontal line at the bottom that was 40 pixels wide and 2 pixels high based on this. 

Yes, I've added an example like that in the test page.

> Then what happens with the 2px wide clause related to contrast - would I not need to meet contrast because it's 2px wide or because it's not the 2px order I still need to meet contrast?

You wouldn't need to meet 'adjacent' contrast with the component, but you still need to meet 'change of contrast' (and non-text contrast with the background from non-text contrast).

I'm not particularly happy with adding this (from a user point of view), I'd rather stick with the straight 1px perimeter. However, it reduces some of the more obscure false positives.

**Comment by alastc** [comment]
Hi everyone,

Based on an update which was surveyed/discussed in the last two weeks, we've updated the size metric to read:

<hr>

**Minimum area**: The area of the focus indicator is either:

- at least as large as the area of a 1 CSS pixel thick perimeter of the unfocused control;
- at least as large as a 4 CSS pixels border along the shortest side, and no thinner than 2 CSS pixels.

<hr>

Survey: https://www.w3.org/2002/09/wbs/35422/wcag22-focus-appearance-updates2/results#xq1
Minutes: https://www.w3.org/2021/01/19-ag-minutes.html#t02 
Understanding document: https://w3c.github.io/wcag/understanding/focus-appearance-minimum.html (Should update soon)

Based on that we'll close this issue, but please raise a new one if there are problems with the new text.
